common/include/pcl/common/impl/bivariate_polynomial.hpp

```diff
@@ -109,12 +109,12 @@ pcl::BivariatePolynomialT<real>::deepCopy (const pcl::BivariatePolynomialT<real>
     degree = other.degree;
     parameters = new real[getNoOfParameters ()];
   }
-  if (other.gradient_x == NULL)
+  if (!other.gradient_x)
   {
-    delete gradient_x; gradient_x=NULL;
-    delete gradient_y; gradient_y=NULL;
+    delete gradient_x;
+    delete gradient_y;
```

Comment:
I believe yo should still set the variables to `nullptr` after this, as the object still exists.